### ISSUE TITLE
Warning: Cannot update during an existing state transition (such as within `render` or another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to `componentWillMount`.

### ISSUE THREAD

**Issue by ChalomEllezam** [ryandrewjohnson/react-localize-redux#84]
**Do you want to request a _feature_ or report a _bug_?** I want to request a bug

**What is the current behavior?** i got this message in my warning console : "Warning: Cannot update during an existing state transition (such as within `render` or another component's constructor). Render methods should be a pure function of props and state; constructor side-effects are an anti-pattern, but can be moved to `componentWillMount`"

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code. Paste the link to your JSFiddle (https://jsfiddle.net/Luktwrdm/) or CodeSandbox (https://codesandbox.io/s/new) example below:**

**What is the expected behavior?** No problem should appear

**Which versions of `react` and `react-localize-redux` are you using?** react-localize-redux V3, the latest

When i try to install the modules, i got this error in my console.log, here is my code 

index.js file

> 
> import React from 'react';
> import ReactDOM from 'react-dom';
> import './index.css';
> import App from './App';
> import registerServiceWorker from './registerServiceWorker';
> import { Provider } from 'react-redux';
> import { store } from './utils/store';
> import {MuiThemeProvider} from '@material-ui/core/styles';
> import {theme} from './theme';
> import {LocalizeProvider} from 'react-localize-redux';
> import setAuthorizationHeader from './utils/authorization_header';
> import {BrowserRouter as Router} from 'react-router-dom';
> 
> 
> 
> if(localStorage.jwtToken){
>     setAuthorizationHeader(localStorage.jwtToken);
> }
> 
> ReactDOM.render(
>     <Router>
>         <MuiThemeProvider theme={theme}>
>             <LocalizeProvider>
>                 <Provider store={store}>
>                     <App/>
>                 </Provider>
>             </LocalizeProvider>
>         </MuiThemeProvider>
>     </Router>
>     , document.getElementById('root'));
> registerServiceWorker();
> 

app.js file 

import React, { Component } from 'react';
import logo from './logo.svg';
import LoginForm from './components/Login';
import UserInfo from './components/User';
import imagesLibrary from './assets/imageslibrary';
import Header from './components/menu/header';
import {LocalizeProvider, withLocalize } from "react-localize-redux";
import globalTranslations from "./globalTranslations";
import {languages} from "./locales/_Languages";
import Router from './routes/router';

class App extends Component {
    componentWillMount(){
        this.props.initialize({
            languages: [
                { name: "English", code: "en" },
                { name: "French", code: "fr" }
            ],
            translation: globalTranslations,
            options: { renderToStaticMarkup : false }
        });
    }
  render() {
    return (
            <div className="App">
                <Header/>
                <Router/>
            </div>
    );
  }
}

export default withLocalize(App);

Homepage when i want to retrieve a translation 

import React, {Component} from 'react';
import Head from './head';
import {withLocalize, Translate } from 'react-localize-redux';
import globalTranslations from "../../globalTranslations";

class HomePage extends Component {
    render(){
        console.log(this.props);
        return(
            <div>
                <Head/>
                <p><Translate id="movie.title" defaultLanguage="en"/></p>
            </div>
        );
    }
}

export default withLocalize(HomePage);

**Comment by ryandrewjohnson** [comment]
@thchia at first glance this looks good. Good point on the `renderChildren` function being called in render, as that is a bad pattern. Also I agree this design is much more clean and follows the "React" way more closely. One thing though .. in order to avoid this being a breaking change we should still explicitly export `Translate` instead of the default export in your example. 

So maybe something like the below example. Also we will need to update both the flow and typescript definitions for `Translate` as well. Let me know if you need any help with that.

```javascript
class TranslateComponent extends Component {

  ...
 
 componentDidUpdate(prevProps) {
    if (this.props.id !== prevProps.id) {
      this.setDefaultTranslation(this.props.context)
    }
  }

  render() {
    return this.renderChildren() // <--- Pure, no side effects
  }
}

export const Translate = (props: TranslateProps) => <Consumer>
  {context => <Translate {...props} context={context} />}
</Consumer>
```

**Comment by thchia** [comment]
@ryandrewjohnson in the PR (#88) I've done just that (there's also a better explanation of the issue and the proposed changes). I called the "old" `Translate` "WrappedTranslate".

As far as I can tell, the props and children stay the same. The only difference is that "WrappedTranslate" has an additional `context` prop, but this is passed internally as it were, not by the user, so I think the public facing API remains the same.

**Comment by thchia** [comment]
Fixed in #88.

**Comment by nicolanilomartino** [comment]
@ChalomEllezam  actually wrote: "I want to request a bug"
You're inviting trouble bro. Usually we request a fix. not a bug. LOL!
